src/decode-ethernet.c

```diff
@@ -48,6 +50,9 @@ int DecodeEthernet(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
         return TM_ECODE_FAILED;
     }
 
+    if (unlikely(p->ethh != NULL) && ((uint8_t *)p->ethh) - p->ext_pkt > MAX_ETH_OFFSET) {
```

Comment:
shouldn't use `p->ext_pkt`, only use `GET_PKT_DATA(p)`.

This solution is also too limiting I think, we should probably count the total number of layers so that too much recursion can't happen anywhere

Reply:
> we should probably count the total number of layers

Can we add a field to the structure `Packet` ?

Or do we need to get into the trouble fo https://github.com/OISF/suricata/pull/5811#discussion_r568441311 ?